gflownet/envs/tree.py

```diff
@@ -1001,53 +1217,127 @@ def get_parents(
                 attributes[Attribute.THRESHOLD] = -1
 
                 parents.append(parent)
-                actions.append((ActionType.PICK_THRESHOLD, -1))
+                if self.continuous:
+                    actions.append((ActionType.PICK_THRESHOLD, -1, -1))
+                else:
+                    threshold_idx = np.where(
+                        np.isclose(
+                            self.thresholds, state[k][Attribute.THRESHOLD].item()
+                        )
+                    )[0].item()
```

Comment:
1. If anything, it would avoid the inner call (the outer is necessary anyway I think, since we need to go from the tensor returned by `torch.where` to an `int` that can be used to construct an action); but that would require converting `self.thresholds` to a tensor - and it's being used in a couple of different places with direct access, so I am not sure if that would even be more efficient.
2. It definitely should! And was in all of my experiments.